tests/queries/0_stateless/02496_format_datetime_in_joda_syntax.reference

```diff
@@ -0,0 +1,231 @@
+-- { echoOn }
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'G');
+AD
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'GG');
+AD
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'GGG');
+AD
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'C');
+20
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'CC');
+20
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'CCC');
+020
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'Y');
+2018
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'YY');
+18
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'YYY');
+2018
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'YYYY');
+2018
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'YYYYY');
+02018
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'e');
+2
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'ee');
+02
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'E');
+Tue
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'EE');
+Tue
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'EEE');
+Tue
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'EEEE');
+Tuesday
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'y');
+2018
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'yy');
+18
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'yyy');
+2018
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'yyyy');
+2018
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'yyyyy');
+02018
+SELECT formatDateTimeInJodaSyntax(toDateTime('2018-01-02 22:33:44'), 'D');
```

Comment:
What happens for `2018-01-15` and `D`? Does `15` get truncated to `5` / is an exception thrown? Note that according to Databricks docs for "Number" ("If the count of letters is one, then the value is output using the minimum number of digits and without padding."), we should return `15` in that case. Let's add a test.

Reply:
already included. 

```
WITH
    '2018-01-12 22:33:44' AS s,
    toDateTime(s) AS datetime,
    toDateTime64(s, 6) AS datetime64,
    toDate(s) AS date,
    toDate32(s) AS date32
SELECT
    formatDateTimeInJodaSyntax(datetime, 'D'),
    formatDateTimeInJodaSyntax(datetime64, 'D'),
    formatDateTimeInJodaSyntax(date, 'D'),
    formatDateTimeInJodaSyntax(date32, 'D')

Query id: 0f2b9fcd-1236-4b21-893a-55370cd8ac50

┌─formatDateTimeInJodaSyntax(datetime, 'D')─┬─formatDateTimeInJodaSyntax(datetime64, 'D')─┬─formatDateTimeInJodaSyntax(date, 'D')─┬─formatDateTimeInJodaSyntax(date32, 'D')─┐
│ 12                                        │ 12                                          │ 12                                    │ 12                                      │
└───────────────────────────────────────────┴─────────────────────────────────────────────┴───────────────────────────────────────┴─────────────────────────────────────────┘

1 rows in set. Elapsed: 0.002 sec. 
```